ovn_k8s/processor/policy_processor.py

```diff
@@ -28,34 +30,192 @@ def __init__(self, pool):
         self._pool = pool
         self._np_watcher_threads = {}
 
-    def _process_ns_event(self, event):
+    def _process_ns_event(self, event, affected_pods, pod_events):
+        log.dbg("Processing event %s from namespace %s" % (
+            event.event_type, event.source))
         namespace = event.source
         # TODO(salv-orlando): handle namespace isolation change events
+
+        def scan_pods():
+            ns_pods = kubernetes.get_pods_by_namespace(
+                variables.K8S_API_SERVER, event.source)
+            if not ns_pods:
+                return
+            for pod in ns_pods:
+                pod_id = pod['metadata']['uid']
+                affected_pods[pod_id] = pod
+                pod_events.setdefault(pod_id, []).append(event)
+
         if event.event_type == 'ADDED':
             log.dbg("Namespace %s added - spawning policy watcher" % namespace)
             watcher_thread = pw.create_namespace_watcher(
                 namespace, self._pool)
             self._np_watcher_threads[namespace] = watcher_thread
+            # Upon restart this event will be receive for existing pods.
+            # The namespace isolation status might have changed while the
+            # watcher was not running. Pods in the namespace need to be
+            # checked again
+            scan_pods()
         elif event.event_type == 'DELETED':
             watcher_thread = self._np_watcher_threads.pop(namespace)
             pw.remove_namespace_watcher(watcher_thread)
+        elif event.event_type == 'MODIFIED':
+            # This a transition in the namespace isolation status. All the pods
+            # in the namespace are affected
+            scan_pods()
+
+    def _process_pod_event(self, event, affected_pods, pod_events):
+        log.dbg("Processing event %s from pod %s" % (
+            event.event_type, event.source))
+        pod_data = event.metadata
+        # Pods are affected only if the namespace is isolated
+        if not kubernetes.is_namespace_isolated(
+                variables.K8S_API_SERVER, pod_data['metadata']['namespace']):
+            log.dbg("Namespace %s for pod %s is not isolated, no further "
```

Comment:
there is actually a chance the processing done after the log stmt is wasteful.
I will double check and if necessary push and additional PR to amend - adding todo comments for now.